### PR TITLE
feat: [std] implement { Array, TypedArray, StaticArray, String }#at

### PR DESCRIPTION
**Pull request by MaxGraey**
proposal (Stage 3):
https://github.com/tc39/proposal-relative-indexing-method

- [x] I've read the contributing guidelines

### CODE REVIEW

File: tests/compiler/std/string.ts [hunk 1]
```diff
@@ -13,6 +13,11 @@ assert("\xDFab" == "ßab");
 
 assert(str.length == 16);
 assert(str.charCodeAt(0) == 0x68);
+assert(str.at(15) == str.charAt(15));
+assert(str.at(-1) == str.charAt(str.length - 1));
+assert(str.at(-str.length) == "h");
+assert(str.at(str.length) == "");
```
Comment (dcodeIO) [inline tests/compiler/std/string.ts:19]: Wouldn't this one produce `undefined` in JS, and should throw in AS?

Comment (MaxGraey) [reply tests/compiler/std/string.ts:19]: Yes, in JS it produce [`undefined`](https://tc39.es/proposal-relative-indexing-method/#sec-string.prototype.at). So you suggest throw an error?

Comment (dcodeIO) [reply tests/compiler/std/string.ts:19]: I believe these are different. `.charAt` returns an empty string in JS for example, but `operator[]` would return `undefined`. So I guess both `operator[]` and `.at` should throw, but `.charAt` remain as-is.

Comment (MaxGraey) [reply tests/compiler/std/string.ts:19]: I see. But this will minor breaking changes

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**